### PR TITLE
#816

### PR DESCRIPTION
**Pull request by wsbrenk**
1. changes proposed in this pull request:

  wip:
  
refactor Special event table

@tychobrailleur  what to do concerning icons:
- replace png by svg icons
- create a getIcon that can handle both kinds (current solution doesn't work)

TODO: 
- [x] icon handling
- [x] layout save and restore 

 `src/main/resources/release_notes.md` ...

 - [x] has been updated
 - [ ] does not require update

### CODE REVIEW

File: src/main/java/core/gui/comp/entry/ColorLabelEntry.java [hunk 1]
```diff
@@ -236,15 +236,39 @@ private void setGraphicalChangeValue(double number) {
...
     }
 
+    private void initAdd() {
+        var isInitDone = this.getLayout() != null && this.getLayout().getClass() == BoxLayout.class;
+        if (!isInitDone){
+            this.setLayout(new BoxLayout(this, BoxLayout.X_AXIS));
+            var icon = this.getIcon();
+            if (icon!=null){
+                super.setIcon(null);
+                this.add(new JLabel(icon));
```
Comment (tychobrailleur) [inline src/main/java/core/gui/comp/entry/ColorLabelEntry.java:246]: Doesn't this get called one time too many the first time in `addIcon`?  I must admit I don't follow the logic. :)

**Comment by tychobrailleur** [comment]
The approach we had decided on with akasolace was to convert to SVG, much easier to deal with.

**Comment by tychobrailleur** [comment]
There is a lot of commented out code that can be removed!

File: src/main/java/core/gui/comp/entry/ColorLabelEntry.java (hunk 1 above)
Comment (wsbrenk) [reply src/main/java/core/gui/comp/entry/ColorLabelEntry.java:246]: this should handle the case if setIcon was called before addIcon is called. This icon will be added to the list of components. The added icon is then appended. (that's the plan)

**Comment by wsbrenk** [comment]
currently I'm looking for svg replacements to the existing png icons. Unfortunately, Hattrick itself still uses PNG icons and e.g. the Rotate.png is nowhere to be found as SVG.

**Comment by wsbrenk** [comment]
next commit will remove the comments.

**Pull request merged**